### PR TITLE
MM-14481: Do not allow to edit or delete in archived channels

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Check it the channel is archived before post actions like edit and delete.

#### Ticket Link
[MM-14481](https://mattermost.atlassian.net/browse/MM-14481)

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-server/blob/master/i18n/en.json)) updates

### CODE REVIEW

File: i18n/en.json [hunk 1]
```diff
@@ -1448,6 +1448,18 @@
     "id": "api.post.create_post.town_square_read_only",
     "translation": "This channel is read-only. Only members with permission can post here."
   },
+  {
+    "id": "api.post.patch_post.can_not_udpate_post_in_deleted.error",
+    "translation": "Can not update a post in a deleted channel."
+  },
+  {
+    "id": "api.post.udpate_post.can_not_udpate_post_in_deleted.error",
```
Comment (mkraft) [inline i18n/en.json]: Typo in the translation key name `udpate` (in 2 places). Also, could `api.post.udpate_post.can_not_udpate_post_in_deleted.error` and `api.post.patch_post.can_not_udpate_post_in_deleted.error` simply share the same translation?

Comment (jespino) [reply i18n/en.json]: Yeah, I going to fix it right away

Comment (jespino) [reply i18n/en.json]: Fixed!

**Comment by jespino** [comment]
I'm not a big fan of reusing translation strings, in this case the translation is equal, but it is true by coincidence, for example, I could change the string to "Partially Update" instead of "Update", and then, one of the translations would be wrong. I think I prefer to have both translations here.

**Pull request merged**